open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/common/idempotency/IdempotencyValidator.java

```diff
@@ -80,123 +79,146 @@ public static IdempotencyValidationResult validateIdempotency(String idempotency
-            return new ArrayList<>();
+    private IdempotencyValidationResult validateIdempotencyConditions(ConsentManageData consentManageData,
+                                                                      DetailedConsentResource consentRequest)
+            throws IdempotencyValidationException, IOException {
+        // Compare the client ID sent in the request and client id retrieved from the database
+        // to validate whether the request is received from the same client
+        if (IdempotencyValidationUtils.isClientIdsMatching(consentManageData.getClientId(),
+                consentRequest.getClientID())) {
+            // Check whether difference between two dates is less than the configured time
+            if (IdempotencyValidationUtils.isRequestReceivedWithinAllowedTime(getCreatedTimeOfPreviousRequest(
+                    consentManageData.getRequestPath(), consentRequest.getConsentID()))) {
+                // Compare whether JSON payloads are equal
+                if (isPayloadSimilar(consentManageData, getPayloadOfPreviousRequest(
+                        consentManageData.getRequestPath(), consentRequest.getConsentID()))) {
+                    log.debug("Payloads are similar and request received within allowed" +
+                            " time. Hence this is a valid idempotent request");
+                    return new IdempotencyValidationResult(true, true,
+                            consentRequest, consentRequest.getConsentID());
+                } else {
+                    log.error(IdempotencyConstants.ERROR_PAYLOAD_NOT_SIMILAR);
+                    throw new IdempotencyValidationException(IdempotencyConstants
+                            .ERROR_PAYLOAD_NOT_SIMILAR);
+                }
+            } else {
+                log.error(IdempotencyConstants.ERROR_AFTER_ALLOWED_TIME);
+                throw new IdempotencyValidationException(IdempotencyConstants
+                        .ERROR_AFTER_ALLOWED_TIME);
+            }
+        } else {
+            log.error(IdempotencyConstants.ERROR_MISMATCHING_CLIENT_ID);
+            throw new IdempotencyValidationException(IdempotencyConstants.ERROR_MISMATCHING_CLIENT_ID);
         }
     }
 
     /**
-     * Method to compare the client ID sent in the request and client id retrieved from the database.
+     * Method to get the Idempotency Attribute Name store in consent Attributes.
      *
-     * @param requestClientID     Client ID sent in the request
-     * @param dbClientId          client ID retrieved from the database
-     * @return   Whether JSON client Ids are equal
+     * @param resourcePath     Resource Path
+     * @return idempotency Attribute Name.
      */
-    private static boolean isClientIdsMatching(String requestClientID, String dbClientId) {
-
-        return requestClientID.equals(dbClientId);
+    public String getIdempotencyAttributeName(String resourcePath) {
+        return "IdempotencyKey";
```

Comment:
Constant?

Reply:
Fixed in https://github.com/wso2/financial-open-banking/pull/51/commits/e1205691228fc4b76853a0eab7dd99dac2516661